From 2883687752d1d84fc79e209d1b22b4bcb7acfadb Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Thu, 16 Jul 2015 18:03:09 +0200
Subject: [PATCH 02/26] check for wayland egl 1.0.0

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 209b929..807cac8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -745,7 +745,7 @@ case $host in
     LIBS=$old_LIBS
     CFLAGS=$old_CFLAGS
 
-    PKG_CHECK_MODULES(WAYLAND_EGL, wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0, HAVE_WAYLAND_EGL=yes, HAVE_WAYLAND_EGL=no)
+    PKG_CHECK_MODULES(WAYLAND_EGL, wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0.0, HAVE_WAYLAND_EGL=yes, HAVE_WAYLAND_EGL=no)
   ;;
 esac
 
-- 
1.9.1

